extras/ai-battle/HeadlessGame.cpp

```diff
@@ -0,0 +1,264 @@
+// Copyright (C) 2005 - 2024 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "HeadlessGame.h"
+#include "EventManager.h"
+#include "GlobalGameSettings.h"
+#include "PlayerInfo.h"
+#include "Savegame.h"
+#include "factories/AIFactory.h"
+#include "network/PlayerGameCommands.h"
+#include "world/GameWorld.h"
+#include "world/MapLoader.h"
+#include "gameTypes/MapInfo.h"
+#include "gameData/GameConsts.h"
+#include <boost/nowide/iostream.hpp>
+#include <chrono>
+#include <cstdio>
+#include <sstream>
+#ifdef WIN32
+#    include "Windows.h"
+#endif
+
+std::vector<PlayerInfo> GeneratePlayerInfo(const std::vector<AI::Info>& ais);
+std::string ToString(const std::chrono::milliseconds& time);
+std::string HumanReadableNumber(unsigned num);
+
+namespace bfs = boost::filesystem;
+namespace bnw = boost::nowide;
+using bfs::canonical;
+
+#ifdef WIN32
+HANDLE setupStdOut();
+void printConsole(const char* fmt, ...);
+#elif defined(__GNUC__)
+void printConsole(const char* fmt, ...) __attribute__((format(printf, 1, 2)));
+#else
+void printConsole(const char* fmt, ...);
+#endif
```

Comment:
Only Jenkins errors with the warning:
> /source/extras/ai-battle/HeadlessGame.cpp:253:64: error: function 'void printConsole(const char*, ...)' might be a candidate for 'gnu_printf' format attribute [-Werror=suggest-attribute=format]

We are using MinGW as the cross-compiler and hence GCC, so this can be reproduced on godbolt: https://godbolt.org/z/sjx4ooMGW

This should work:
```suggestion
#ifdef WIN32
HANDLE setupStdOut();
#endif

#ifdef __GNUC__
void printConsole(const char* fmt, ...) __attribute__((format(printf, 1, 2)));
#else
void printConsole(const char* fmt, ...);
#endif
```